pkg/compose/pull.go

```diff
@@ -291,6 +291,22 @@ func (s *composeService) pullRequiredImages(ctx context.Context, project *types.
 	})
 }
 
+func isServiceImageToBuild(service types.ServiceConfig, services []types.ServiceConfig) bool {
+	if service.Build != nil {
+		return true
+	}
+	if service.DependsOn != nil {
```

Comment:
I wonder we should require `depends_on` here (which has undefined behavior regarding build)
could just check all other services in the compose file for missing image

Reply:
Right we can check if the `image` name is defined in a service with a `build` section